### ISSUE TITLE
How is this improved?

### ISSUE THREAD

**Issue by Sicily-F** [kozistr/improved-edge-connect#1]
Hi there,

I am very interested in using this repo, and I was wondering if you could provide an explanation of how it is the new improved version? I've been struggling to get Edge-Connect to work on my own data for a while.

Any tips would be great!

**Comment by kozistr** [comment]
First of all, thanks for your interest : )

Actually, It's been a long time while I work on it, so I can't remember everything, and maybe the whole works weren't uploaded to this repo because it's related to the company work.

Anyway, what I tried were
1. perceptual loss (style loss part in this repo) : [code](https://github.com/kozistr/improved-edge-connect/blob/master/src/loss.py#L46)
  * Based on ESRGAN, when calculating perceptual loss, It's better to use the output of the convolution layer instead of the output of the activation layer (you can find the details in the paper). So, I tested it on my data and it works better!
2. network modules : [code](https://github.com/kozistr/improved-edge-connect/blob/master/src/networks.py)
  * I adopted Instance Layer Normalization (a.k.a ILN), and several techniques I guess.
3. custom mask
  * I didn't upload the code, which makes a mask, I made a custom mask logic, which fits in my task, and it gives a good result.

That's all I have tried! You can apply other techniques and SOTA image-inpainting architecture (2021).
If you have any questions, feel free to leave a comment : )

Thanks in advance!
Kim

**Comment by Sicily-F** [comment]
Hi Kim,


Thank you so much for your response - this is great and so quick!!!

I will definitely try those things for sure. Can I ask if you could share your config.yml file for training on your own data? I would like to use the pretrained model, but train it further on my data, in an application of transfer learning. This is the tricky part for me!


Many thanks,
Sicily

**Comment by kozistr** [comment]
Unfortunately, I don't have any settings & models now because it related to a company project a few years ago : (
Maybe  you can train your model on the big dataset (like CelebA, Place365, etc) first, then fine-tune on your dataset!
